src/server/middleware/security/robot2.go

```diff
@@ -0,0 +1,82 @@
+package security
+
+import (
+	"fmt"
+	"github.com/goharbor/harbor/src/common/security"
+	robotCtx "github.com/goharbor/harbor/src/common/security/robot"
+	"github.com/goharbor/harbor/src/common/utils"
+	robot_ctl "github.com/goharbor/harbor/src/controller/robot"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/permission/types"
+
+	"github.com/goharbor/harbor/src/pkg/robot2/model"
+	"net/http"
+	"strings"
+)
+
+type robot2 struct{}
+
+func (r *robot2) Generate(req *http.Request) security.Context {
+	log := log.G(req.Context())
+	name, secret, ok := req.BasicAuth()
+	if !ok {
+		return nil
+	}
+	if !strings.HasPrefix(name, config.RobotPrefix()) {
+		return nil
+	}
+	key, err := config.SecretKey()
+	if err != nil {
+		log.Error("failed to get secret key")
+		return nil
+	}
+	_, err = utils.ReversibleDecrypt(secret, key)
+	if err != nil {
+		log.Errorf("failed to decode secret key: %s, %v", secret, err)
+		return nil
```

Comment:
Do we need to return an unauthorized security context here? Because we can confirm that the user tries to access the harbor with a robot account but provide a wrong password.